src/main/java/fridgy/model/ingredient/BaseIngredient.java

```diff
@@ -0,0 +1,95 @@
+package fridgy.model.ingredient;
+
+import java.util.Objects;
+
+import fridgy.commons.util.CollectionUtil;
+import fridgy.model.base.Eq;
+
+/**
+ * Represents a Base ingredient in the Inventory, to be used by Recipe.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class BaseIngredient implements Eq {
+
+    // Identity fields
+    private final Name name;
+    private final Quantity quantity;
+
+    /**
+     * Constructs a base ingredient reference with the given name and quantity.
+     */
+    public BaseIngredient(Name name, Quantity quantity) {
+        CollectionUtil.requireAllNonNull(name, quantity);
```

Comment:
This seems unneccessarily verbose. May be just import requireAllNonNull instead of the class that holds it?